src/Module/Api/Mastodon/FollowRequests.php

```diff
@@ -78,41 +83,23 @@ public static function rawContent(array $parameters = [])
 
 		$baseUrl = DI::baseUrl();
 
-		if (isset($since_id) && isset($max_id)) {
-			$condition = ['`uid` = ? AND NOT `ignore` AND `id` > ? AND `id` < ?', self::$current_user_id, $since_id, $max_id];
-		} elseif (isset($since_id)) {
-			$condition = ['`uid` = ? AND NOT `ignore` AND `id` > ?', self::$current_user_id, $since_id];
-		} elseif (isset($max_id)) {
-			$condition = ['`uid` = ? AND NOT `ignore` AND `id` < ?', self::$current_user_id, $max_id];
-		} else {
-			$condition = ['`uid` = ? AND NOT `ignore`', self::$current_user_id];
-		}
-
-		$count = DBA::count('intro', $condition);
-
-		$intros = DBA::selectToArray(
-			'intro',
-			[],
-			$condition,
-			['order' => ['id' => 'DESC'], 'limit' => $limit]
+		$Introductions = DI::intro()->selectByBoundaries(
+			['`uid` = ? AND NOT `ignore`', self::$current_user_id],
+			['order' => ['id' => 'DESC']],
+			$since_id,
+			$max_id,
+			$limit
 		);
 
 		$return = [];
-		foreach ($intros as $intro) {
-			$cdata = Contact::getPublicAndUserContacID($intro['contact-id'], $intro['uid']);
-			if (empty($cdata['public'])) {
-				continue;
-			}
 
-			$publicContact = Contact::getById($cdata['public']);
-			$userContact = Contact::getById($cdata['user']);
-			$apcontact = APContact::getByURL($publicContact['url'], false);
-			$account = Mastodon\Account::create($baseUrl, $publicContact, $apcontact, $userContact);
-
-			// Not ideal, the same "account" can have multiple ids depending on the context
-			$account->id = $intro['id'];
-
-			$return[] = $account;
+		foreach ($Introductions as $key => $Introduction) {
```

Comment:
Question: How does this work?

The `DI::intro()->selectByBoundaries` returns a "global" `Collections` model, if you don't override it, isn't it? But how is than the `Model` inside the `Collections` type-safe?